engine/server/engine/centralized_logs/client_implementations/persistent_volume/stream_logs_strategy/per_week_stream_logs_strategy.go

```diff
@@ -162,8 +162,20 @@ func (strategy *PerWeekStreamLogsStrategy) getRetainedLogsFilePaths(
 	enclaveUuid, serviceUuid string) []string {
 	var paths []string
 
-	// get log file paths as far back as they exist
+	// scan for first existing log file
+	firstWeekWithLogs := 0
 	for i := 0; i < (retentionPeriodInWeeks + 1); i++ {
+		year, week := strategy.time.Now().Add(time.Duration(-i) * oneWeek).ISOWeek()
+		filePathStr := fmt.Sprintf(volume_consts.PerWeekFilePathFmtStr, volume_consts.LogsStorageDirpath, strconv.Itoa(year), strconv.Itoa(week), enclaveUuid, serviceUuid, volume_consts.Filetype)
+		if _, err := filesystem.Stat(filePathStr); err == nil {
```

Comment:
@tedim52 is there a difference between a file not existing and a "real" error happening during `stat`? If there is we may want to handle the `err != nil` case as well

Reply:
Yep, good to handle in case. Done.